jax/experimental/host_callback.py

```diff
@@ -591,20 +604,30 @@ def _raise_if_using_outfeed_with_pjrt_c_api(backend: xb.XlaBackend):
 XlaLocalClient = xla_client.Client
 DType = Any
 
+class CallbackFlavor(enum.Enum):
+  """Specifies which flavor of callback to use under JAX_HOST_CALLBACK_LEGACY=False.
+
+  See https://github.com/google/jax/issues/20385."""
+  IO_CALLBACK = 1  # uses jax.experimental.io_callback
+  PURE = 2  # uses jax.pure_callback
+  DEBUG = 3  # uses jax.debug.callback, valid only when there are no results
+
 
 def _deprecated_id_tap(tap_func,
            arg,
            *,
            result=None,
            tap_with_device=False,
            device_index=0,
+           callback_flavor=CallbackFlavor.IO_CALLBACK,
```

Comment:
Is there a reason for making this a parameter instead of always passing `CallbackFlavor.IO_CALLBACK` to `_call`?